### PR TITLE
UI.PerformAudioPassThru

### LINKED ISSUE #401: Feature Additions For Missing UI Components
Each of these features could be added to the Generic HMI natively. 
Missing UI Components: 

- UI.Slider RPC 
- Scrollable Message
- UI.ClosePopUp RPC
- BasicCommunication.DecryptCertificate
- Consent Prompts & SDL.GetUserFriendlyMessage 
- TTS.StopSpeaking
- UI.PerformAudioPassThru

### PR DESCRIPTION
**Pull request by JackLivio**
Fixes #401 (Partially)

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Core Tests
 - Send Perform Audio Pass Thru RPC
 - Test different buttons send correct result codes
 - Send an EndAudioPassThru to end the prompt early.
 
Core version / branch / commit hash / module tested against: develop
Proxy+Test App name / version / branch / commit hash / module tested against: rpcb-js

### Summary
Implements the popup needed for UI.PerformAudioPassThru. There are 3 static buttons that have different results (Done, retry, cancel). This pr also implements UI.EndAudioPassThru

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Review by iCollin — COMMENTED** [review]
I only reviewed commit https://github.com/smartdevicelink/generic_hmi/pull/404/commits/b90ec81f971818b044b81a9b1ce44cda9c5446f7

We should also add these text fields to each template: `audioPassThruDisplayText1`, `audioPassThruDisplayText2`.

File: src/js/AppHeader.js [hunk 1]
```diff
@@ -162,6 +172,22 @@ class AppHeader extends React.Component {
                         buttons={this.props.softButtons}
                         appName={this.props.scrollableMessageAppName}/>
                 </Modal>
+                <Modal
+                isOpen={this.props.showPerformAudioPassThru}
+                className={'app-body alertModal'}
+                overlayClassName={`${themeClass} alertOverlay`}
+                contentLabel="Example Modal"
+                onRequestClose={this.closeAudioPassThru}
```
[the remark this answers is not in the document]
Comment (JackLivio) [reply src/js/AppHeader.js:180]: > this.closeAudioPassThru = this.closeAudioPassThru.bind(this)

This shouldnt be needed when using the arrow function syntax.

File: src/js/AppHeader.js [hunk 2]
```diff
@@ -162,6 +172,22 @@ class AppHeader extends React.Component {
...
+                >
+                    <PerformAudioPassThru
+                        theme={this.props.theme}
+                        textFields={this.props.aptTextFields}
+                        appName={this.props.aptAppName}
+                        resultCallback={(result) => {
+                            this.closeAudioPassThru(result)
+                        }}
```
[the remark this answers is not in the document]
Comment (JackLivio) [reply src/js/AppHeader.js:188]: Leaving the arrow syntax so i can pass the appropriate result code to the method

**Comment by JackLivio** [comment]
Addressed comments

File: src/js/AppHeader.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (iCollin) [reply src/js/AppHeader.js]: this change is totally optional, but resultCallback will be called with result if you apply my suggestion too. adding this extra function just wraps closeAudioPassThru with a proxy that doesn't really do anything.

resultCallback(result)
(result) => {}
closeAudioPassThru(result)

**Review by iCollin — APPROVED** [review]

**Pull request merged**